pkg/controller/acmechallenges/finalizer.go

```diff
@@ -32,16 +32,15 @@ import (
 // finalizerRequired returns true if the finalizer is not found on the challenge.
 //
 // API transition
-// We currently only add cmacme.ACMELegacyFinalizer, but a future version will add
-// cmacme.ACMEDomainQualifiedFinalizer.
+// We only add cmacme.ACMEDomainQualifiedFinalizer.
 // A finalizer only needs to be added if neither is present.
 func finalizerRequired(ch *cmacme.Challenge) bool {
 	finalizers := sets.NewString(ch.Finalizers...)
-	return !finalizers.Has(cmacme.ACMELegacyFinalizer) &&
-		!finalizers.Has(cmacme.ACMEDomainQualifiedFinalizer)
+	return !finalizers.Has(cmacme.ACMEDomainQualifiedFinalizer) &&
+		!finalizers.Has(cmacme.ACMELegacyFinalizer)
```

Comment:
```suggestion
	return !finalizers.Has(cmacme.ACMEDomainQualifiedFinalizer)
```

Looking at the use of this function, I think this should be changed now.